### PR TITLE
VZ-5953 CLI Uninstall Command error handling

### PR DESCRIPTION
**Pull request by stevshap**
Fixes VZ-5953

Adds error handling to prevent uninstall logs from writing to a closed channel.

### CODE REVIEW

File: tools/vz/cmd/helpers/vpo.go [hunk 1]
```diff
@@ -223,26 +226,30 @@ func WaitForOperationToComplete(client clipkg.Client, kubeClient kubernetes.Inte
 	}(vzHelper.GetOutputStream())
 
 	// goroutine to wait for the completion of the operation
-	go func() {
+	go func(outputStream io.Writer) {
```
Comment (mgianatagh) [inline tools/vz/cmd/helpers/vpo.go:229]: why was outputStream added to the function?  I don't see it being used.

File: tools/vz/cmd/uninstall/uninstall.go [hunk 2]
```diff
@@ -227,34 +246,51 @@ func waitForUninstallToComplete(client client.Client, kubeClient kubernetes.Inte
...
+	defer close(resChan)
+
+	feedbackChan := make(chan bool)
+	defer close(feedbackChan)
+
+	go func(outputStream io.Writer) {
 		sc := bufio.NewScanner(rc)
 		sc.Split(bufio.ScanLines)
 		for sc.Scan() {
 			_, _ = fmt.Fprintf(vzHelper.GetOutputStream(), fmt.Sprintf("%s\n", sc.Text()))
```
Comment (mgianatagh) [inline tools/vz/cmd/uninstall/uninstall.go:260]: this line should be using the variable outputStream

File: tools/vz/cmd/uninstall/uninstall.go [hunk 3]
```diff
@@ -227,34 +246,51 @@ func waitForUninstallToComplete(client client.Client, kubeClient kubernetes.Inte
...
-			vz, err := helpers.GetVerrazzanoResource(client, namespacedName)
-			if vz == nil {
-				resChan <- nil
-			}
-			if err != nil && !errors.IsNotFound(err) {
-				resChan <- err
+		}
+	}(vzHelper.GetOutputStream())
+
+	go func(outputStream io.Writer) {
```
Comment (mgianatagh) [inline tools/vz/cmd/uninstall/uninstall.go]: outputStream is not used by this func

File: tools/vz/cmd/uninstall/uninstall_test.go [hunk 4]
```diff
@@ -147,13 +149,16 @@ func TestUninstallCmdDefaultTimeout(t *testing.T) {
...
 	assert.NotNil(t, cmd)
-	_ = cmd.PersistentFlags().Set(constants.TimeoutFlag, "2ns")
+	_ = cmd.PersistentFlags().Set(constants.TimeoutFlag, "2ms")
 
 	// Run upgrade command
 	err := cmd.Execute()
 	assert.NoError(t, err)
 	assert.Equal(t, "", errBuf.String())
-	assert.Contains(t, buf.String(), "Timeout 2ns exceeded waiting for uninstall to complete")
+	// This timeout is so short because the vz resource gets deleted from cache almost instantaneously.
```
Comment (mgianatagh) [inline tools/vz/cmd/uninstall/uninstall_test.go:158]: Now that you moved the 1 second pause to the top of the wait loop, I think this comment can say that the timeout needs to be less than the one second polling delay due to the vz getting deleted instantly.  For the fun of it I locally modified the timeout to be 950ms and ran the test a few times, and it passed.

**Comment by stevshap** [comment]
@mgianatagh resolved feedback

File: tools/vz/cmd/uninstall/uninstall_test.go (hunk 4 above)
Comment (stevshap) [reply tools/vz/cmd/uninstall/uninstall_test.go:158]: ok will update

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**